File: Sources/Project Templates/Fire/watchOS Extension.xctemplate/SceneKit_InterfaceController.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"
#import "GameController.h"

@interface ___FILEBASENAME___ ()

@property (strong, nonatomic) IBOutlet WKInterfaceSCNScene *scnInterface;
@property (strong, nonatomic) GameController *gameController;

@end

@implementation ___FILEBASENAME___

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    
    self.gameController = [[GameController alloc] initWithSceneRenderer:self.scnInterface];
}

- (IBAction)handleTap:(WKTapGestureRecognizer *)gestureRecognize {
    // Highlight the tapped nodes
    CGPoint p = gestureRecognize.locationInObject;
    [self.gameController highlightNodesAtPoint:p];
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}

@end

File: Sources/Project Templates/Fire/watchOS Extension.xctemplate/SceneKit_InterfaceController.swift
//___FILEHEADER___

import WatchKit
import SceneKit

class ___FILEBASENAME___: WKInterfaceController {
    
    @IBOutlet var scnInterface: WKInterfaceSCNScene!
    var gameController: GameController!
    
    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        gameController = GameController(sceneRenderer: scnInterface)
    }
    
    func handleTap(_ gestureRecognizer: WKTapGestureRecognizer) {
        // Highlight the tapped nodes
        let p = gestureRecognizer.locationInObject()
        gameController.highlightNodes(atPoint: p)
    }
    
    override func willActivate() {
        // This method is called when watch view controller is about to be visible to user
        super.willActivate()
    }
    
    override func didDeactivate() {
        // This method is called when watch view controller is no longer visible
        super.didDeactivate()
    }
    
}
